src/kanalytics.js

```diff
@@ -223,20 +223,18 @@ export default class KAnalytics extends BasePlugin {
    * @return {void}
    */
   _sendAnalytics(eventType: number): void {
-    let statsEvent = new Event(eventType);
+    const statsEvent = new Event(eventType);
     statsEvent.currentPoint = this.player.currentTime;
     statsEvent.duration = this.player.duration;
     statsEvent.seek = this._hasSeeked;
     statsEvent.hasKanalony = this.config.hasKanalony;
     Object.assign(statsEvent, this._playerParams);
-
-    let request: RequestBuilder = StatsService.collect(this.config.playerVersion, this._ks, {"event": statsEvent}, this.config.baseUrl);
+    const request: RequestBuilder = OVPStatsService.collect(this.config.serviceUrl, this._ks, this.config.playerVersion, {"event": statsEvent});
```

Comment:
most of params don't change until media is changed, so why can't we init a service with `URL`, `KS` and `playerVersion` and then just call collect with `{"event": statsEvent}`?

Reply:
Good point, but this is require design change in the provider side. Lets do it after the current refactor.